src/EditorFeatures/CSharpTest/EditAndContinue/RudeEditStatementTests.cs

```diff
@@ -905,6 +905,529 @@ public void MatchExceptionHandlers()
+                { "return (d, 6);", "return (d, 5);" },
+                { "return (10, e, 22);", "return (10, e);" },
+                { "return (2, () => {      int a = 6;     return 1; });", "return (2, () => {     int a = 6;     return 5; });" },
+                { "() => {      int a = 6;     return 1; }", "() => {     int a = 6;     return 5; }" },
+                { "{      int a = 6;     return 1; }", "{     int a = 6;     return 5; }" },
+                { "int a = 6;", "int a = 6;" },
+                { "int a = 6", "int a = 6" },
+                { "a = 6", "a = 6" },
+                { "return 1;", "return 5;" }
+            };
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        public void MatchLocalFunctionDefinitions()
+        {
+            var src1 = @"
+(int a, string c) F1(int i) { return null; }
+(int a, int b) F2(int i) { return null; }
+(int a, int b, int c) F3(int i) { return null; }
+";
+
+            var src2 = @"
+(int a, int b) F1(int i) { return null; }
+(int a, int b, string c) F2(int i) { return null; }
+(int a, int b) F3(int i) { return null; }
+";
+
+            var match = GetMethodMatches(src1, src2, kind: MethodKind.Regular);
+            var actual = ToMatchingPairs(match);
+
+            var expected = new MatchingPairs
+            {
+                { "(int a, string c) F1(int i) { return null; }", "(int a, int b) F1(int i) { return null; }" },
+                { "{ return null; }", "{ return null; }" },
+                { "return null;", "return null;" },
+                { "(int a, int b) F2(int i) { return null; }", "(int a, int b, string c) F2(int i) { return null; }" },
+                { "{ return null; }", "{ return null; }" },
+                { "return null;", "return null;" },
+                { "(int a, int b, int c) F3(int i) { return null; }", "(int a, int b) F3(int i) { return null; }" },
+                { "{ return null; }", "{ return null; }" },
+                { "return null;", "return null;" }
+            };
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        public void MatchVariableDesignations()
+        {
+            var src1 = @"
+M(out int z);
+N(out var a);
+";
+
+            var src2 = @"
+M(out var z);
+N(out var b);
+";
+
+            var match = GetMethodMatches(src1, src2, kind: MethodKind.Regular);
+            var actual = ToMatchingPairs(match);
+
+            var expected = new MatchingPairs
+            {
+                { "M(out int z);", "M(out var z);" },
+                { "z", "z" },
+                { "N(out var a);", "N(out var b);" },
+                { "a", "b" }
+            };
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        public void MatchParenthesizedVariable_Update()
+        {
+            var src1 = @"
+var (x1, (x2, x3)) = (1, (2, true));
+var (a1, a2) = (1, () => { return 7; });
+";
+
+            var src2 = @"
+var (x1, (x2, x4)) = (1, (2, true));
```

Comment:
>x2 [](start = 10, length = 2)

It'd be interesting here to change this to (x1, (x3, x4)) to check that x3 is matched.